src/Configuration/src/SpringBoot/SpringBootEnvironmentVariableProvider.cs

```diff
@@ -60,13 +61,7 @@ public override void Load()
 
             foreach (string key in keys)
             {
-                string? value = Data[key];
-
-                if (key.Contains('.') && value != null)
-                {
-                    string newKey = key.Replace('.', ':');
-                    Data[newKey] = value;
-                }
+                Data[ConfigurationKeyConverter.AsDotNetConfigurationKey(key)] = Data[key];
```

Comment:
Same concern here: I wouldn't want to suggest escaping/arrays/etc can be used in environment variables, unless there's a use case.

Reply:
I think this provider is related to Spring Cloud Data Flow, which may or may not even be relevant anymore... Apparently the docs were never written for either of the spring boot config providers - I'm inclined to leave these with the shared method since this one should have already had support for `__` too but I don't think that would have worked before